worldedit-core/src/main/java/com/fastasyncworldedit/core/function/pattern/TypeSwapPattern.java

```diff
@@ -30,24 +33,36 @@ public class TypeSwapPattern extends AbstractExtentPattern {
      * @param extent       extent to use
      * @param inputString  string to replace. May be regex.
      * @param outputString string to replace with
+     * @param allowRegex   if regex should be allowed for input string matching
      * @since TODO
      */
-    public TypeSwapPattern(Extent extent, String inputString, String outputString) {
+    public TypeSwapPattern(Extent extent, String inputString, String outputString, boolean allowRegex) {
         super(extent);
         this.inputString = inputString;
         this.outputString = outputString;
         if (!StringMan.isAlphanumericUnd(inputString)) {
-            this.inputPattern = Pattern.compile(inputString);
+            if (allowRegex) {
+                this.inputPattern = Pattern.compile(inputString.replace(",", "|"));
+                inputs = null;
+            } else {
+                inputs = SPLITTER.split(inputString);
+            }
+        } else {
+            inputs = null;
         }
     }
 
     @Override
     public boolean apply(Extent extent, BlockVector3 get, BlockVector3 set) throws WorldEditException {
         BlockState existing = get.getBlock(extent);
         String oldId = existing.getBlockType().getId();
-        String newId;
+        String newId = oldId;
         if (inputPattern != null) {
             newId = inputPattern.matcher(oldId).replaceAll(outputString);
+        } else if (inputs != null && inputs.length > 0) {
+            for (String input : inputs) {
+                newId = oldId.replace(input, outputString);
```

Comment:
I assume this should be 
```suggestion
                newId = newId.replace(input, outputString);
```
?

Reply:
Also that handling is missing in the other methods, can this somehow be moved into one method to avoid inconsistencies?

Reply:
The for loop? This is the only case where there's anything to loop over

Reply:
But why? What's the difference between `apply` and the `applyBlock` methods?

Reply:
Ah okay I see what you mean, had forgotten about the other one